lib/internal/bootstrap/loaders.js

```diff
@@ -208,6 +210,81 @@
         NativeModule.require;
       fn(this.exports, requireFn, this, process);
 
+      if (config.experimentalModules) {
+        this.namespace = Object.keys(this.exports);
+
+        const update = (property, value) => {
+          if (this.reflect !== undefined && this.namespace.includes(property))
+            this.reflect.exports[property].set(value);
+        };
+
+        const methodWrapMap = new WeakMap();
+
+        const wrap = (target, name, value) => {
+          if (typeof value !== 'function')
+            return value;
+
+          if (typeof value.name === 'string' && /^bound /.test(value.name))
+            return value;
+
```

Comment:
Besides non-functions and bound-functions you can also skip non-native functions.
I have an inference method [here](https://github.com/standard-things/esm/blob/3.0.25/src/util/is-native.js).

Reply:
we can't skip native functions. for instance if you do `module.exports = new Map()` and we don't wrap the exports then `Map.prototype.*` will have improper receivers and throw

Reply:
so the only functions we bind then are unbound native functions? what about member functions written in js

Reply:
> so the only functions we bind then are unbound native functions? what about member functions written in js

We aren't binding functions. The wrapper juggles the `thisArg` around for the one, maybe two, cases that cause native methods grief but beyond that we forward the `thisArg` along. So it only makes sense to wrap the methods that need the `thisArg` juggling in the first place (native methods).